src/corelibs/U2View/src/ov_msa/MaEditor.h

```diff
@@ -69,6 +69,31 @@ class SNPSettings {
     QVariantMap highlightSchemeSettings;
 };
 
+/** Set of row ordering modes supported by MaEditor. */
+enum class MaEditorRowOrderMode {
+    /**
+     * The default order of sequences in the MA: same as in the original source file.
+     * Note: there are no collapsible groups support in this mode.
+     */
+    Original,
+
+    /**
+     * Rows are ordered by sequence content similarity.
+     * Sequences with the equal content are joined into collapsible groups.
+     * This mode is managed/supported internally by MA editor: MA editor automatically rebuilds groups when MA content changes.
+     * The order is not saved to the file and is kept in the view memory only.
+     * This mode allows to re-order read-only MA objects.
+     */
+    Sequence,
+
+    /**
+     * Sequences are ordered by some external manager (e.g. tree-view) and can be re-ordered by user (e.g. drag & drop).
+     * The order is not saved to the file and is kept in the view memory only.
+     * This mode allows to re-order read-only MA objects.
+     */
+    Free
```

Comment:
Is it possible to distinguish between these cases (tree-view and drag & drop)?

Reply:
There is no need today to differentiate between any externally managed row ordering modes in the MSA Editor, so it does not matter if this is a tree guided sort or any other kind of sort injected by a 3rd party plugin.
In the Free mode MSA editor should not save changes to the file (the mode is virtual) and should not dynamically recompute sequence content similarity groups.

Once we will have a use case that need the differentiation we will support it.